### DISCUSSION TITLE
Debian installatie

### DISCUSSION THREAD

**Discussion by iwanVgeelen** [minvws/nl-kat-coordination#11] [Q&A] [answered]
In de wiki van openkat wordt voor de Debian installatie de url [https://kat.rdosecurity.nl/debian/](https://kat.rdosecurity.nl/debian/%22%20%5Ct%20%22_blank) gegeven.
De documentatie lijkt te suggereren dat je hier handmatig de betreffende packages moet downloaden, terwijl als je op de url klikt het een Debian repository lijkt te zijn.
Is deze repository al gevuld en zo ja, wat zou ik moeten toevoegen aan de APT sources?

**Comment by ErikNoort** [comment]
Kan iemand een update geven wanneer er Debian packages beschikbaar gemaakt gaan worden?
Alvast bedankt!

**Comment by underdarknl** [comment]
@ErikNoort we hopen de debian packages een dezer dagen beschikbaar te kunnen te stellen. Ik zal hier een berichtje plaatsen als we zover zijn. Ook de container images voor Docker komen deze release beschikbaar. 
We zoeken uiteraard ook mensen die alvast onze huidige builds willen testen. Zou jij daarvoor beschikbaar zijn?

**Reply by ErikNoort** [reply]
Daarvoor ben ik zeker beschikbaar. Ik heb geen ervaring met het zelf packagen voor Debian, maar gebruik al wel heel lang Debian, dus het toevoegen van een custom apt source en daarna installeren zou geen probleem moeten zijn.
De VM met Debian staat al een tijdje klaar :-)

**Comment by RobboRob** [comment, marked as answer]
Ervaringen met installatie Debian packages:

Uitgangspunt:
Debian distributie 1.15.0 64 bits
CPU: 1vCPU
RAM 4 GB (ook met 8GB geprobeerd)
HD: 32GB

indien nodig repro's toevoegen in sources.list:

_sudo vi /etc/apt/sources.list_
deb http://deb.debian.org/debian bullseye main
deb http://deb.debian.org/debian bullseye-updates main

Daarna _sudo apt update_ uitvoeren en de installatie starten

Installatie Debian packages uitvoeren volgens instructies op [de Debian installatie pagina](https://github.com/minvws/nl-kat-coordination/discussions/11):

Voor de upgrade naar de nieuwe versie (Debian package versie 0.0.4 alias openKAT 1.3.0) heb ik de installatie stappen enigszins uitgebreid:
VERSION=0.0.4
rm -f kat-debian-packages*.gz.*
rm -rf kat-debian-packages
wget "https://rdosecurity.nl/hqxjpworLJXazhub2RwqZl0cmbVKCclt5nw3HmX5LB/kat-debian-packages_${VERSION}.tar.gz"
tar -xzvf kat-debian-packages_${VERSION}.tar.gz
cd kat-debian-packages_${VERSION}
sudo apt install -y --no-install-recommends ./kat-*_${VERSION}_all.deb
sudo systemctl daemon-reload
sudo systemctl restart kat-rocky kat-mula kat-bytes kat-boefjes kat-normalizers kat-katalogus kat-keiko kat-octopoes kat-octopoes-worker

Ëventueel opvragen errors:
_sudo journalctl -n 100 -u kat-mula|grep -i error_

NOOT:
Voor versie 0.0.4 voor RabbitMQ toevoegen aan /etc/rabbitmq/rabbitmq.conf
listeners.tcp.local6 = ::1:5672

**Reply by ErikNoort** [reply]
top dat er nu Debian packages beschikbaar zijn.
Komt er ook nog een apt source beschikbaar? Dan gaan de (security) updates tenminste automatisch.
Ik ga volgende week een installatie uitvoeren.

**Comment by RobboRob** [comment]
ook nog uitgevoerd zojuist om te zorgen dat alles na een reboot weer opstart:

systemctl enable kat-rocky kat-mula kat-bytes kat-boefjes kat-normalizers kat-katalogus kat-keiko kat-octopoes kat-octopoes-worker

**Comment by RobboRob** [comment]
Twee dingen werken nu nog niet:

1) health pagina (krijg een Server Error (500))
2) pdf's genereren (krijg een pdf document waarin in tekst alleen maar _None_ staat.

**Reply by ErikNoort** [reply]
Ik heb succesvol OpenKAT geinstalleerd op een Debian VM met de verstrekte packages.
Ik hoop wel dat er ook nog een APT source beschikbaar komt.

Issues:
alle download PDF knoppen geven een server error (500)

**Reply by underdarknl** [reply]
> Ik heb succesvol OpenKAT geinstalleerd op een Debian VM met de verstrekte packages. Ik hoop wel dat er ook nog een APT source beschikbaar komt.
> 
> Issues: alle download PDF knoppen geven een server error (500)

Issue met de pdf's is gefixt in de volgende release, en uiteraard willen wij netjes een APT repository beschikbaar stellen.

**Comment by underdarknl** [comment]
Er zijn nieuwe packages beschikbaar onder de 1.4.RC1 release tag. Als jullie die weer kunnen testen is dat tof!
https://github.com/minvws/nl-kat-coordination/releases/tag/v1.4.0rc1
